xds/internal/balancer/outlierdetection/balancer.go

```diff
@@ -596,13 +606,35 @@ func (b *outlierDetectionBalancer) Target() string {
 // if the SubConn is not ejected.
 func (b *outlierDetectionBalancer) handleSubConnUpdate(u *scUpdate) {
 	scw := u.scw
-	scw.latestState = u.state
-	if !scw.ejected {
-		if scw.listener != nil {
-			b.childMu.Lock()
-			scw.listener(u.state)
-			b.childMu.Unlock()
-		}
+	scw.mu.Lock()
+	scw.healthListener = nil
+	scw.mu.Unlock()
+
+	if scw.listener == nil {
+		return
+	}
+
+	if !scw.healthListenerEnabled {
+		scw.latestDeleveredState = u.state
```

Comment:
We are updating this field here, but there is a chance that this update is not delivered to the child policy if the `if` condition below does not return `true`.